Fix intermittent race test

This test is failing intermittently (at least on my MBA M3):
```
<nil>
&{{0 0 <nil>} {0 0 <nil>} false  <nil> 0xc00050c780 0x10067e1a0 0 0 0xc00019a600 test 0xc000520000 0xc00018f080 false {{} {{} {} 0} 0} 0xc0005112d0 0xc000511340 0xc0005113b0 <nil> {{} {0 0}} {{} {{} 0} {{} {0 0}}} {{} {{} 0} {{} {0 0}}} {{} {{} 0} {{} {0 0}}} {{} {{} 0} {{} {0 0}}}}
--- FAIL: TestStreamTailWithFollow (0.01s)
    --- FAIL: TestStreamTailWithFollow/with_follow_data,_past_data_arrives_before_and_after (0.00s)
        stream_test.go:921:
                Error Trace:    /Users/andres/worktrees/glorious-istanbul/modules/shared/logs/stream_test.go:921
                Error:          Not equal:
                                expected: []string{"s1-b", "s2-b", "s1-c", "s2-c"}
                                actual  : []string{"s1-b", "s2-b", "s2-c", "s1-c"}

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -3,4 +3,4 @@
                                  (string) (len=4) "s2-b",
                                - (string) (len=4) "s1-c",
                                - (string) (len=4) "s2-c"
                                + (string) (len=4) "s2-c",
                                + (string) (len=4) "s1-c"
                                 }
                Test:           TestStreamTailWithFollow/with_follow_data,_past_data_arrives_before_and_after
FAIL
FAIL    github.com/kubetail-org/kubetail/modules/shared/logs    1.152s
```

--- a/modules/shared/logs/stream.go
+++ b/modules/shared/logs/stream.go
@@ -511,6 +511,28 @@ LOOP:
 		}
 	}
 
+	// Drain any pending future events that were queued before the past stream
+	// finished. This ensures they are included in the sorted buffer below.
+	drainCtx, cancelDrainCtx := context.WithTimeout(s.rootCtx, 50*time.Millisecond)
+	defer cancelDrainCtx()
+
+DrainLoop:
+	for {
+		select {
+		case <-s.rootCtx.Done():
+			return // exit
+		case <-drainCtx.Done():
+			break DrainLoop
+		case r, ok := <-s.futureCh:
+			if !ok {
+				break DrainLoop
+			}
+			buffer = append(buffer, r)
+		default:
+			break DrainLoop
+		}
+	}
+
 	// Exit if not following
 	if !s.follow {
 		return

